install_config/upgrading/migrating_etcd.adoc

```diff
@@ -0,0 +1,278 @@
++
+To check the etcd cluster is healthy you can run:
++
+----
+# etcdctl <certificate_details> <endpoint> cluster-health
+member 2a3d833935d9d076 is healthy: got healthy result from https://etcd-test-1:2379
+member a83a3258059fee18 is healthy: got healthy result from https://etcd-test-2:2379
+member 22a9f2ddf18fee5f is healthy: got healthy result from https://etcd-test-3:2379
+cluster is healthy
+----
++
+To check a difference of raft indices you can run:
++
+----
+# ETCDCTL_API=3 etcdctl <certificate_details> <endpoint> -w table endpoint status
++------------------+------------------+---------+---------+-----------+-----------+------------+
+|     ENDPOINT     |        ID        | VERSION | DB SIZE | IS LEADER | RAFT TERM | RAFT INDEX |
++------------------+------------------+---------+---------+-----------+-----------+------------+
+| etcd-test-1:2379 | 2a3d833935d9d076 | 3.1.9   | 25 kB   | false     |       415 |        995 |
+| etcd-test-2:2379 | a83a3258059fee18 | 3.1.9   | 25 kB   | true      |       415 |        995 |
+| etcd-test-3:2379 | 22a9f2ddf18fee5f | 3.1.9   | 25 kB   | false     |       415 |        995 |
++------------------+------------------+---------+---------+-----------+-----------+------------+
+----
++
+If the minimum and maximum of raft indexes over all etcd members differ by more
+than one unit, wait a minute and try the command again.
+
+. +++<b>Migrate and scale up etcd:</b>+++
++
+[WARNING]
+====
+The migration should not be run repeatedly, as new v2 data can overwrite v3 data
+that has already migrated.
+====
+
+.. Stop etcd on all etcd hosts:
++
+----
+# systemctl stop etcd
+----
+
+.. Running the following command (with the *etcd* daemon stopped) on your first
+etcd host to perform the migration:
++
+----
+# ETCDCTL_API=3 etcdctl migrate --data-dir=/var/lib/etcd
+----
++
+The `--data-dir` target can in a different location depending on the deployment.
+For example, embedded etcd operates over the
+*_/var/lib/origin/openshift.local.etcd_* directory, and etcd run as a system
+container operates over the *_/var/lib/etcd/etcd.etcd_* directory.
++
+When complete, the migration responds with the following message if successful:
++
+----
+finished transforming keys
+----
++
+If there is no v2 data, it responds with:
++
+----
+no v2 keys to migrate
+----
+
+.. If the `etcdctl migrate` command is run without the `--no-ttl` option, TTL keys
+are migrated as well. Given that the TTL keys in v2 data are replaced with leases in
+v3 data, you must attach leases to all migrated TTL keys by running the following (with
+*etcd* daemon running):
+
+... Create a new cluster on the first host:
++
+----
+# echo "ETCD_FORCE_NEW_CLUSTER=true" >> /etc/etcd/etcd.conf
+# systemctl start etcd
+# sed -i '/ETCD_FORCE_NEW_CLUSTER=true/d' /etc/etcd/etcd.conf
+# systemctl restart etcd
+----
+
+... Scale up additional etcd hosts by following the
+xref:/backup_restore.adoc#adding-addtl-etcd-members[Adding Additional etcd
+Members] documentation.
```

Comment:
@sdodson Should this "Scale up..." and the following "After your etcd cluster..." steps be substeps of the "If the etcd migrate command" step? Or should these 2 substeps be pulled up a level?

Reply:
Both "Create a new" and "Scale up..."  should be run before the doc mentions the TTL migration and should be on the same level as the "If the etcdctl migrate command"

Reply:
Rearranged, thanks!